### PR TITLE
tests for opensearch webhook were implemented

### PR DESCRIPTION
**Pull request by tengu-alt**

### CODE REVIEW

**Review by ribaraka — APPROVED** [review]

**Review by worryg0d — APPROVED** [review]

File: apis/clusters/v1beta1/validation.go [hunk 1]
```diff
@@ -216,6 +216,30 @@ func validateTwoFactorDelete(new, old []*TwoFactorDelete) error {
...
 }
 
+func validateIngestNodes(new, old []*OpenSearchIngestNodes) error {
+	if len(old) != len(new) {
+		return models.ErrImmutableIngestNodes
+	}
+
+	if *old[0] != *new[0] {
+		return models.ErrImmutableIngestNodes
+	}
```
Comment (testisnullus) [inline apis/clusters/v1beta1/validation.go:226]: Do we validate that only 1 or fewer items can be appended to the IngestNodes array while creation?

File: apis/clusters/v1beta1/validation.go [hunk 2]
```diff
@@ -216,6 +216,30 @@ func validateTwoFactorDelete(new, old []*TwoFactorDelete) error {
...
+	return nil
+}
+
+func validateClusterManagedNodes(new, old []*ClusterManagerNodes) error {
+	if len(old) != len(new) {
+		return models.ErrImmutableClusterManagedNodes
+	}
+
+	if *old[0] != *new[0] {
+		return models.ErrImmutableClusterManagedNodes
```
Comment (testisnullus) [inline apis/clusters/v1beta1/validation.go:237]: Same is here

File: apis/clusters/v1beta1/opensearch_webhook_test.go [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+		err = yaml.Unmarshal(yfile, &openSearchManifest)
+		Expect(err).Should(Succeed())
+		openSearchManifest.Spec.TwoFactorDelete = []*TwoFactorDelete{{Email: "emailTEST", Phone: "phoneTEST"}}
+		testOpenSearchManifest = openSearchManifest
+	})
+
+	ctx := context.Background()
+
+	When("apply a OpenSearch manifest", func() {
```
Comment (testisnullus) [inline apis/clusters/v1beta1/opensearch_webhook_test.go:30]: an Opensearch

File: apis/clusters/v1beta1/opensearch_webhook_test.go [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+
+			testOpenSearchManifest.Spec.DataCentres[0].CloudProvider = "GCP"
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+			testOpenSearchManifest.Spec.DataCentres[0] = &prevDC
+
+			prevStringValue := openSearchManifest.Spec.DataCentres[0].ProviderAccountName
+			testOpenSearchManifest.Spec.DataCentres[0].ProviderAccountName = models.DefaultAccountName
+			Expect(k8sClient.Create(ctx, &testOpenSearchManifest)).ShouldNot(Succeed())
+			testOpenSearchManifest.Spec.DataCentres[0].ProviderAccountName = prevStringValue
```
Comment (testisnullus) [inline apis/clusters/v1beta1/opensearch_webhook_test.go:67]: Do we plan to add tests for RIYOA accounts?

RIYOA - run in your own account. It means, that customer will run Instaclustr Products using their own AWS/AZURE/GCP accounts.

**Review by testisnullus — CHANGES_REQUESTED** [review]

File: apis/clusters/v1beta1/opensearch_webhook_test.go (hunk 3 above)
Comment (tengu-alt) [reply apis/clusters/v1beta1/opensearch_webhook_test.go:30]: fixed

File: apis/clusters/v1beta1/opensearch_webhook_test.go (hunk 4 above)
Comment (tengu-alt) [reply apis/clusters/v1beta1/opensearch_webhook_test.go:67]: this test case is made for ensure that  cloudProviderSettings is available only for RIYOA accounts

**Review by testisnullus — APPROVED** [review]

**Pull request merged**